src/TextInput.js

```diff
@@ -0,0 +1,62 @@
+// @flow
+import React from "react";
+import { View, TextInput, StyleSheet } from "react-native";
+
+import Text from "./Text";
+import { TextStyle } from "./styles";
+
+const defaultStyles = StyleSheet.create({
+  // reset default padding of android device.
+  textInput: {
+    padding: 0,
+  },
+  extraText: {
+    alignSelf: "flex-start",
+  },
+});
+
+function makeExtraText(
+  error?: string | null,
+  errorStyle?: TextStyle,
+  optionText?: string,
+  optionTextStyle?: TextStyle
+): Text | null {
+  if (error) {
+    return <Text style={[defaultStyles.extraText, errorStyle]}>{error}</Text>;
+  } else if (optionText) {
+    return (
+      <Text style={[defaultStyles.extraText, optionTextStyle]}>
+        {optionText}
+      </Text>
+    );
+  } else if (error !== undefined) {
```

Comment:
This condition is very complex. What do you mean by error is not undefined? What is the different between error is null and error is undefined?
I think a better practice will be we set default value of error to null and dont use undefined.

Reply:
I prefer `!= null`

Reply:
If the error is undefined, that means user not provide error prop, so this component won't create nonvisible Text.
If the error is null, that means the user does provide error prop, so the component will create nonvisible Text to avoid layout unstable.

Reply:
Then the type of `error` should be `error: string | null | undefined`. This makes the user to think what should they provide. `error?: string | null` means the prop is optional.

However, I still think it is better always reserve space for the error text.

Reply:
Maybe we can provide the prop named `reserveErrorSpace`, this may help thing not so confused.